src/tqec/noise_models/noise_model.py

```diff
@@ -0,0 +1,570 @@
+            flip_result: The probability that a measurement result should be reported incorrectly.
+                Only valid when applied to operations that produce measurement results.
+        """
+        if not (0 <= flip_result <= 1):
+            raise ValueError(f"not (0 <= {flip_result=} <= 1)")
+        for k, p in after.items():
+            if OP_TYPES[k] != NOISE:
+                raise ValueError(f"not a noise channel: {k} from {after=}")
+            if not (0 <= p <= 1):
+                raise ValueError(f"not (0 <= {p} <= 1) from {after=}")
+        self.after = after
+        self.flip_result = flip_result
+
+    def append_noisy_version_of(
+        self,
+        *,
+        split_op: stim.CircuitInstruction,
+        out_during_moment: stim.Circuit,
+        after_moments: defaultdict[tuple[str, float], stim.Circuit],
+        immune_qubits: AbstractSet[int],
+    ) -> None:
+        targets = split_op.targets_copy()
+        if immune_qubits and any(
+            (t.is_qubit_target or t.is_x_target or t.is_y_target or t.is_z_target)
+            and t.value in immune_qubits
+            for t in targets
+        ):
+            out_during_moment.append(split_op)
+            return
+
+        args = split_op.gate_args_copy()
+        if self.flip_result:
+            t = OP_TYPES[split_op.name]
+            assert t == MPP or t == JUST_MEASURE_1Q or t == MEASURE_RESET_1Q
+            assert len(args) == 0
+            args = [self.flip_result]
+
+        out_during_moment.append(split_op.name, targets, args)
+        raw_targets = [t.value for t in targets if not t.is_combiner]
+        for op_name, arg in self.after.items():
+            after_moments[(op_name, arg)].append(op_name, raw_targets, arg)
+
+
+class NoiseModel:
+    def __init__(
+        self,
+        idle_depolarization: float,
+        additional_depolarization_waiting_for_m_or_r: float = 0,
+        gate_rules: dict[str, NoiseRule] | None = None,
+        measure_rules: dict[str, NoiseRule] | None = None,
+        any_clifford_1q_rule: NoiseRule | None = None,
+        any_clifford_2q_rule: NoiseRule | None = None,
+    ):
+        self.idle_depolarization = idle_depolarization
+        self.additional_depolarization_waiting_for_m_or_r = (
+            additional_depolarization_waiting_for_m_or_r
+        )
+        self.gate_rules = gate_rules
+        self.measure_rules = measure_rules
+        self.any_clifford_1q_rule = any_clifford_1q_rule
+        self.any_clifford_2q_rule = any_clifford_2q_rule
+
+    @staticmethod
+    def si1000(p: float) -> "NoiseModel":
+        """Superconducting inspired noise.
+
+        As defined in "A Fault-Tolerant Honeycomb Memory":
+        https://arxiv.org/abs/2108.10457
+
+        Small tweak when measurements aren't immediately followed by a reset: the measurement result
+        is probabilistically flipped instead of the input qubit. The input qubit is depolarized after
+        the measurement.
```

Comment:
Oh sorry for my misunderstanding. What I mean is that we do not need the depolarizing error on the measured qubits after the measurement. As for using probabilistic flip of measurement results instead of actual flipping the qubits before measurement, I don't think it will make difference on the simulation results of our circuits.

Reply:
So right now I do not have a deep-enough understanding to have a good opinion on that. I'll agree with whatever seems reasonable to you. Could you submit a suggestion if anything needs to change?